deletion-policy/models.ts

```diff
@@ -0,0 +1,93 @@
+//
+// README: Ensure this is in sync with the models defined in the api package
+//
+
+import { EventEnvironment } from "bog-analytics";
+import mongoose, { Schema, Types } from "mongoose";
+
+export interface BaseEvent extends Document {
+  category: string;
+  subcategory: string;
+  projectId: string | Types.ObjectId;
+  environment: EventEnvironment;
+  createdAt?: Date | string;
+  updatedAt?: Date | string;
+}
+
+export interface Project extends Document {
+  _id: string | Types.ObjectId;
+  clientApiKey: string;
+  serverApiKey: string;
+  projectName: string;
+  privateData: boolean;
+  deletionPolicy: number;
+  createdAt?: Date;
+  updatedAt?: Date;
+}
+
+export interface BaseEventWithoutCategory extends Document {
+  projectId: string | Types.ObjectId;
+  environment: EventEnvironment;
+  createdAt?: Date | string;
+  updatedAt?: Date | string;
+}
+
+export interface CustomEvent extends BaseEventWithoutCategory {
+  eventTypeId: string | Types.ObjectId;
+  properties: object;
+}
```

Comment:
Something like this would be better:

```
import { BaseEvent, Project, CustomEvent } from 'bog-analytics';
import { Document } from 'mongoose';

export interface BaseEventInterface extends ImportedBaseEvent, Document {}
export interface ProjectInterface extends ImportedProject, Document {}
export interface CustomEventInterface extends ImportedCustomEvent, Document {}
```

That way you don't need to keep these types explicitly in sync.